### PR TITLE
feat(ssi): add submit endpoints

### PR DESCRIPTION
**Pull request by Phil91**
## Description

add endpoints to submit usecaseparticipation and ssi certificates

## Why

To be able to submit use case participations a endpoint was needed, same for an ssi certificate

## Issue

N/A - Jira Issue: CPLP-2854

## Corresponding CD PR

[#34](https://github.com/eclipse-tractusx/portal-cd/pull/34)

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Phil91** [comment]
waiting for #84 to be merged, since this PR builds upon the code

**Comment by Phil91** [comment]
> #### ⚠️ GitGuardian has uncovered 2 secrets following the scan of your pull request.
> Please consider investigating the findings and remediating the incidents. Failure to do so may lead to compromising the associated services or software components.
> 
> 🔎 Detected hardcoded secrets in your pull request
> 🛠 Guidelines to remediate hardcoded secrets
> 🦉 [GitGuardian](https://dashboard.gitguardian.com/auth/login/?utm_medium=checkruns&utm_source=github&utm_campaign=cr1) detects secrets in your source code to help developers and security teams secure the modern development process. You are seeing this because you or someone else with access to this repository has authorized GitGuardian to scan your pull request.Our GitHub checks need improvements? [Share your feedbacks](https://050127dayho.typeform.com/to/KmeAPTMk)!

The finding are only uuid's used for tests, don't know why gitGuardian is marking them as security issue (secret).
I've already shared my feedback

File: src/administration/Administration.Service/BusinessLogic/CompanyDataBusinessLogic.cs [hunk 1]
```diff
@@ -237,4 +243,70 @@ await _portalRepositories
                 )))
             .ToListAsync()
             .ConfigureAwait(false);
+
+    /// <inheritdoc />
+    public async Task CreateUseCaseParticipation((Guid UserId, Guid CompanyId) identity, UseCaseParticipationCreationData data, CancellationToken cts)
```
Comment (ntruchsess) [inline src/administration/Administration.Service/BusinessLogic/CompanyDataBusinessLogic.cs:248]: please rename the CancellationToken to 'cancellationToken' as it is being done in all the standard apis. (do so in all other methods that have a CancellationToken parameter)

File: src/administration/Administration.Service/BusinessLogic/CompanyDataBusinessLogic.cs [hunk 2]
```diff
@@ -237,4 +243,70 @@ await _portalRepositories
...
+        CancellationToken cts)
+    {
+        if (await companyCredentialDetailsRepository.CheckSsiDetailsExistsForCompany(identity.CompanyId, credentialTypeId, kindId, verifiedCredentialExternalTypeDetailId).ConfigureAwait(false))
+        {
+            throw new ControllerArgumentException("Credential request already existing");
+        }
+
+        var (documentContent, hash) = await document.GetContentAndHash(cts).ConfigureAwait(false);
+        var doc = _portalRepositories.GetInstance<IDocumentRepository>().CreateDocument(document.FileName, documentContent,
+            hash, document.ContentType.ParseMediaTypeId(), DocumentTypeId.PRESENTATION, x =>
```
Comment (ntruchsess) [inline src/administration/Administration.Service/BusinessLogic/CompanyDataBusinessLogic.cs:296]: as the mediaTypeId is already being parsed be the caller it shall be passed to the method as parameter.

File: src/marketplace/Apps.Service/BusinessLogic/AppChangeBusinessLogic.cs [hunk 3]
```diff
@@ -158,7 +158,7 @@ private static async Task<IEnumerable<LocalizedDescription>> ValidateAndGetAppDe
     public async Task UploadOfferAssignedAppLeadImageDocumentByIdAsync(Guid appId, (Guid UserId, Guid CompanyId) identity, IFormFile document, CancellationToken cancellationToken)
     {
         var appLeadImageContentTypes = new[] { MediaTypeId.JPEG, MediaTypeId.PNG };
-        var documentContentType = ContentTypeMapperExtensions.ParseMediaTypeId(document.ContentType);
+        var documentContentType = MediaTypeIdExtensions.ParseMediaTypeId(document.ContentType);
```
Comment (ntruchsess) [inline src/marketplace/Apps.Service/BusinessLogic/AppChangeBusinessLogic.cs:161]: please use the extension way of calling 'document.ContentType.ParseMediaTypeId()'. Then use the new extension-method 'CheckDocumentContent' to validate.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/administration/Administration.Service/BusinessLogic/CompanyDataBusinessLogic.cs (hunk 1 above)
Comment (Phil91) [reply src/administration/Administration.Service/BusinessLogic/CompanyDataBusinessLogic.cs]: done

File: src/administration/Administration.Service/BusinessLogic/CompanyDataBusinessLogic.cs (hunk 2 above)
Comment (Phil91) [reply src/administration/Administration.Service/BusinessLogic/CompanyDataBusinessLogic.cs]: done

File: src/marketplace/Apps.Service/BusinessLogic/AppChangeBusinessLogic.cs (hunk 3 above)
Comment (Phil91) [reply src/marketplace/Apps.Service/BusinessLogic/AppChangeBusinessLogic.cs]: changed

**Review by ntruchsess — APPROVED** [review]

File: src/portalbackend/PortalBackend.PortalEntities/Entities/UseCaseParticipationStatus.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Enums;
+using System.ComponentModel.DataAnnotations;
+
+namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Entities;
+
+public class UseCaseParticipationStatus
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Entities/UseCaseParticipationStatus.cs:26]: this class doesn't have any references?

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.PortalEntities/Entities/UseCaseParticipationStatus.cs (hunk 4 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Entities/UseCaseParticipationStatus.cs]: that was kinda odd... my rider wasn't showing the file at all. i did remove it.

**Review by ntruchsess — APPROVED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ICompanyCredentialDetailsRepository.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.DBAccess.Models;
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/ICompanyCredentialDetailsRepository.cs:21]: just noticed this file seems to be redundant too.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Models/UseCaseParticipation.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Enums;
+
+namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.DBAccess.Models;
+
+public record UseCaseParticipation
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.DBAccess/Models/UseCaseParticipation.cs:25]: this file seems to be redundant as well. It's only referenced by ICompanyCredentialDetailsRepository.cs (which seems to be redundant).

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Review by ntruchsess — APPROVED** [review]
This PR now seems to be fine, but CD-PR https://github.com/eclipse-tractusx/portal-cd/pull/34 is still draft?

**Comment by Phil91** [comment]
> This PR now seems to be fine, but CD-PR [eclipse-tractusx/portal-cd#34](https://github.com/eclipse-tractusx/portal-cd/pull/34) is still draft?

Sorry i set i to active

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**